From 34af1a5d58c45ab82c2a01034843ac81b6838349 Mon Sep 17 00:00:00 2001
From: Yongqiang Sun <yongqiang.sun@amd.com>
Date: Thu, 14 Jan 2016 11:17:10 -0500
Subject: [PATCH 0694/1110] drm/amd/dal: Fixed active dongle bug.

In case of active dongle downstream unplug,
dc_link_detect doesn't handle correctly.
Add connection type check after detect_dp.

Signed-off-by: Yongqiang Sun <yongqiang.sun@amd.com>
Acked-by: Jordan Lazare <Jordan.Lazare@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc_link.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
index b034c2f..46132f2 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
@@ -608,6 +608,10 @@ bool dc_link_detect(const struct dc_link *dc_link, bool boot)
 				&converter_disable_audio,
 				&audio_support, boot);
 
+			/* Active dongle downstream unplug */
+			if (link->public.type == dc_connection_none)
+				return true;
+
 			if (link->public.type == dc_connection_mst_branch)
 				return false;
 
-- 
2.7.4

